src/stealth/stealth.js

```diff
@@ -48,14 +55,51 @@ export default function applyStealthToBrowser(browser, options) {
 
     contextPrototype.newPage = async function (...args) {
         const page = await prevNewPage.bind(this)(...args);
+        addStealthDebugToPage(page);
         await applyStealthTricks(page, opts);
+
+        await checkNumberOfIrames(page);
         return page;
     };
 
-
     return Promise.resolve(modifiedBrowser);
 }
 
+/**
+ * Logs the stealth errors in browser to the node stdout.
+ * @param page {Page} - puppeteer page instace
+ */
+function addStealthDebugToPage(page) {
+    page.on('console', (msg) => {
+        const text = msg.text();
+
+        if (text.includes(STEALTH_ERROR_MESSAGE_PREFIX)) {
+            log.error(text);
+        }
+    });
+}
+
+/**
+ * Overrides the page.goto function in order to check for iframes after the navigation. If the
+ * @param page {Page} - puppeteer page
+ */
+function checkNumberOfIrames(page) {
+    const prevGoto = page.goto;
+
+    page.goto = async function (...args) {
+        const response = await prevGoto.bind(this)(...args);
```

Comment:
This itself could be a performance overhead. I suggest adding this only if the debug mode is on. The problem with this iframe warning itself is that we do not have the loaded page in general. So the only way I could think of is to override the goto function. Since the newPage is already overridden I think it is not a big deal, however, I am open to suggestions on how to do it better.

Reply:
How about instead of counting the frames, count the number of times stealth was applied in the page and print a warning if it exceeds 10? Is it possible?

Reply:
I could do it through the specific log messages - the same way I do it for the errors. However it could be then detected, but let's just assume that the protections are not actively looking to protect against our stealth and that they probably don't analyze or even collect console dumps.

Reply:
How about using this https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage to send messages and keep a counter in the main frame?